src/js/common/schemaform/Address.jsx

```diff
@@ -42,14 +42,24 @@ class Address extends React.Component {
   }
 
   handleChange(path, update) {
-    let newState = set(path, update, this.props.formData);
+    const formData = this.props.formData
+      ? this.props.formData
+      : getDefaultFormState(this.props.schema, undefined, this.props.registry.definitions);
+
+    let newState = set(path, update, formData);
 
     // if country is changing we should clear the state
     if (path === 'country') {
       newState = set('state', undefined, newState);
     }
 
-    this.props.onChange(newState);
+    const fields = Object.keys(newState);
+    if (fields.length === 0 || fields.every(field => field === 'country' || newState[field] === undefined)) {
```

Comment:
For the second condition, `fields.every(field => field === 'country' || newState[field] === undefined)`, we don't care if `country` has an undefined value because we've prevented the ability to select the blank placeholder option? Shouldn't we just check `newState[field] === undefined` for all fields?